src/rust/engine/src/rule_graph.rs

```diff
@@ -143,6 +143,20 @@ impl Entry {
   }
 }
 
+#[derive(Eq, Hash, PartialEq, Clone, Debug)]
+pub enum SelectorPath {
```

Comment:
Can you add a larger docstring to this one? I think I understand what's going on, but.

Is the idea that this uniquely keys a set of tasks that are possible in some context? And the 'path' bit is that it represents information that used to be represented by multiple Nodes in the graph?